src/rpc/misc.cpp

```diff
@@ -142,6 +143,87 @@ static UniValue createmultisig(const JSONRPCRequest& request)
+              "or more path elements separated by \"/\", where \"h\" represents a hardened child key.\n"
+              "For more information on output descriptors, see the documentation in the doc/descriptors.md file.\n",
+              {
+                {"descriptor", RPCArg::Type::STR, false},
+                {"start",  RPCArg::Type::NUM, true},
+                {"end",  RPCArg::Type::NUM, true}
+              }
+            }.ToString() +
+            "\nArguments:\n"
+            "1. \"descriptor\"                     (string, required) The descriptor\n"
+            "2. \"start\"                          (integer, optional) If a ranged descriptor is used, this specifies the start of the range to import.\n"
+            "3. \"end\"                            (integer, optional) If a ranged descriptor is used, this specifies the end of the range to import.\n"
+            "\nResult:\n"
+            "[ address ] (array) the derived addresses\n"
+            "\nExamples:\n"
+            "\nFirst three native segwit receive addresses\n"
+            + HelpExampleCli("deriveaddresses", "\"wpkh([d34db33f/84h/0h/0h]xpub6DJ2dNUysrn5Vt36jH2KLBT2i1auw1tTSSomg8PhqNiUtx8QX2SvC9nrHu81fT41fvDUnhMjEzQgXnQjKEu3oaqMSzhSrHMxyyoEAmUHQbY/0/*)\" 0 2") +
+            ""
+        );
+
+    RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VNUM, UniValue::VNUM});
+    const std::string desc_str = request.params[0].get_str();
+
+    int range_start = 0;
+    int range_end = 0;
+
+    if (request.params.size() >= 2) {
+        if (request.params.size() == 2) {
+          throw JSONRPCError(RPC_INVALID_PARAMETER, "Missing range end parameter");
+        }
+        range_start = request.params[1].get_int();
+        range_end = request.params[2].get_int();
+    }
+
+    FlatSigningProvider provider;
+    auto desc = Parse(desc_str, provider);
+    if (!desc) {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Invalid descriptor"));
+    }
+
+    if (!desc->IsRange() && request.params.size() > 1) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Range should not be specified for an un-ranged descriptor");
+    }
+
+    if (desc->IsRange() && request.params.size() == 1) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Range must be specified for a ranged descriptor");
+    }
+
+    UniValue addresses(UniValue::VARR);
+
+    for (int i = range_start; i <= range_end; ++i) {
+        std::vector<CScript> scripts;
+        if (!desc->Expand(i, provider, scripts, provider)) {
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Cannot derive script without private keys"));
+        }
+
+        for (const CScript &script : scripts) {
+            CTxDestination dest;
+            if (!ExtractDestination(script, dest)) {
+                throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Descriptor produced invalid address"));
```

Comment:
Not: This condition would appear when the scriptPubKey does not have a corresponding address (I think this is only currently possible through "raw" or through bare multisig). Perhaps a better error message is possible?

Reply:
Could also have a test for this error?

Reply:
If anyone has an example (raw) descriptor that triggers this error, I can add it, and think of a better message.

Reply:
Try "multi(1,key1,key2)" for some public keys.

Reply:
Added the bare multisig example and reworded error to "Descriptor does not have a corresponding address".